galley/cmd/galley/cmd/root.go

```diff
@@ -70,6 +70,17 @@ func GetRootCmd(args []string) *cobra.Command {
 			serverArgs.CredentialOptions.CACertificateFile = validationArgs.CACertFile
 			serverArgs.CredentialOptions.KeyFile = validationArgs.KeyFile
 			serverArgs.CredentialOptions.CertificateFile = validationArgs.CertFile
+			// Default MCPClient credentials to MCP Server if not supplied
+			if serverArgs.MCPClientCredOpts.CertificateFile == "" {
```

Comment:
Our use case is insecure dial-out MCP and outside k8s. To stand-in (for the dial-out MCP) a dynamically configured sidecar & sds is a wish, but not sure when it will happen.